cpplint.py

```diff
@@ -979,26 +1006,38 @@ def ParseNolintSuppressions(filename, raw_line, linenum, error):
     linenum: int, the number of the current line.
     error: function, an error handler.
   """
-  matched = Search(r'\bNOLINT(NEXTLINE)?\b(\([^)]+\))?', raw_line)
+  matched = Search(r'\bNOLINT(NEXTLINE|BEGIN|END)?\b(\([^)]+\))?', raw_line)
   if matched:
-    if matched.group(1):
-      suppressed_line = linenum + 1
+    no_lint_type = matched.group(1)
+    if no_lint_type == 'NEXTLINE':
+      process_category = lambda c: _error_suppressions.setdefault(c, set()).add(linenum + 1)
+    elif no_lint_type == 'BEGIN':
+      process_category = lambda c: _block_error_suppressions.update((c,))
+    elif no_lint_type == 'END':
+      def process_category(category):
+        if _block_error_suppressions[category] <= 0:
```

Comment:
WHat if a user wants to suppress 3 categories, but then just remove those 3 with `NOLINTEND(*)` without repeating them?

Reply:
I can add that functionality easily. It does however break the symmetry between begin and end and it could be problematic for nesting like in this case
```
// NOLINTBEGIN(category1)
...
// NOLINTBEGIN(*)
...
// NOLINTEND(*) <-- Does this end all categories or just the set applied. If the second then the blocks would have to be tracked in a stack
...
// NOLINTEND(category1)
```